### PR TITLE
Implement a nested side panel component

### PR DESCRIPTION
**Pull request by deioo**
Preview: http://g.recordit.co/VQkP4y0tWs.gif
Requires https://github.com/alphasights/paint/pull/64

* Removes js animations and lets css handle everything
* Grabs only the animation duration and triggers events based on it
* Adds sub-components like main and drawer, each with header, footer and content

### CODE REVIEW

File: addon/components/as-side-panel.js [hunk 1]
```diff
@@ -3,20 +3,26 @@ import KeyEventsMixin from 'ember-cli-paint/mixins/key-events';
...
 
-    Ember.$.Velocity(this.$('> div')[0], {
-      right: 0
-    }, {
-      duration: 200
-    });
+  onDidInsertElement: function() {
+    Ember.run.later(this, function() {
+      this.set('isActive', true);
+    }, 1);
```
Comment (Arkham) [inline addon/components/as-side-panel.js]: Does this run after 1 millisecond?

File: addon/components/as-side-panel.js [hunk 2]
```diff
@@ -3,20 +3,26 @@ import KeyEventsMixin from 'ember-cli-paint/mixins/key-events';
...
 
-  onDidInsertElement: function() {
-    this.set('isActive', true);
-    this.set('initialWidth', this.$('> div').width());
+    if (cssDuration.indexOf('ms') !== -1) {
+      return duration;
+    } else {
+      return duration * 1000;
+    }
+  }).volatile(),
```
Comment (Arkham) [inline addon/components/as-side-panel.js]: Can we extract this function and the one in sidebar.js into a mixin?

File: addon/components/as-side-panel.js (hunk 1 above)
Comment (deioo) [inline addon/components/as-side-panel.js]: If the class is added when the page is loaded / element is created, the animation doesn't get triggered.
This seemed like the easiest way to do it.

File: addon/components/as-side-panel.js (hunk 2 above)
Comment (deioo) [inline addon/components/as-side-panel.js]: Btw this is the same behaviour used for the sidebar.

**Comment by eugeniodepalo** [comment]
:+1: :100: :1234: :8ball:

**Pull request merged**